### PR TITLE
narzedzia#3

### PR DESCRIPTION
**Pull request by marcindus**
Team:
marcindus
adamvm

### CODE REVIEW

File: shm/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
+project(shm)
+cmake_minimum_required(VERSION 2.8.2)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+list(APPEND  SHM_SOURCES   Player.cpp Island.cpp Coordinates.cpp Ship.cpp  Alcohol.cpp Fruit.cpp Item.cpp Map.cpp)
+
+ADD_LIBRARY( ${PROJECT_NAME}-lib STATIC ${SHM_SOURCES} )
```
Comment (ziobron) [inline shm/CMakeLists.txt]: we should use lowercase function names

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -14,5 +14,20 @@ jobs:
         chmod +x check.sh
         sudo apt install -y pcregrep
```
Comment (ziobron) [inline .github/workflows/main.yml]:
```suggestion
        chmod +x check.sh
        sudo apt-get update
        sudo apt install -y pcregrep
```

File: shm/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+# dependencies automatically when using CMake 2.8.11 or
+# later. Otherwise we have to add them here ourselves.
+if (CMAKE_VERSION VERSION_LESS 2.8.11)
+  include_directories("${gtest_SOURCE_DIR}/include")
+endif()
+
+enable_testing()
+
+add_executable(${PROJECT_NAME} main.cpp )
+add_executable(${PROJECT_NAME}-ut test.cpp )
```
Comment (ziobron) [inline shm/CMakeLists.txt]: I don't like your formatting with space before a closing parenthesis. It's not uniform with an opening parenthesis.

File: shm/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,55 @@
+project(shm)
+cmake_minimum_required(VERSION 2.8.2)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+list(APPEND  SHM_SOURCES   Player.cpp Island.cpp Coordinates.cpp Ship.cpp  Alcohol.cpp Fruit.cpp Item.cpp Map.cpp)
```
Comment (ziobron) [inline shm/CMakeLists.txt]: Formatting - many spaces around SHM_SOURCES

**Review by ziobron — CHANGES_REQUESTED** [review]
It would be nice if you have some additional compilation flags like `-Wall -Wextra -pedantic -Wconversion -Werror`

File: shm/CMakeLists.txt (hunk 3 above)
Comment (marcindus) [reply shm/CMakeLists.txt]: fixed

File: shm/CMakeLists.txt (hunk 1 above)
Comment (marcindus) [reply shm/CMakeLists.txt]: fixed

File: shm/CMakeLists.txt (hunk 4 above)
Comment (marcindus) [reply shm/CMakeLists.txt]: fixed

**Pull request closed without merging**